trie/iterator.go

```diff
@@ -292,6 +292,44 @@ func (it *nodeIterator) peek(descend bool) (*nodeIteratorState, *int, []byte, er
 	return nil, nil, nil, errIteratorEnd
 }
 
+// peekSeek is like peek, but it also tries to skip resolving hashes by skipping
+// over the siblings that do not lead towards the desired seek position.
+func (it *nodeIterator) peekSeek(seekKey []byte) (*nodeIteratorState, *int, []byte, error) {
+	if len(it.stack) == 0 {
+		// Initialize the iterator if we've just started.
```

Comment:
Please put this stuff into a new method, for example `it.init`, instead of duplicating it here.